### ISSUE TITLE
Sphinx autosummary tables not displaying

### ISSUE THREAD

**Issue by cwrowley** [rtfd/readthedocs.org#1290]
There seems to be some sort of problem with tables using the Sphinx autosummary extension not showing up in the documentation.  For example, I sometimes use tables like this:

```
.. autosummary::
    :toctree: generated/

    foo
    bar
```

This works just fine when I generate docs on my local machine, and used to work on readthedocs.org (as recently as one month ago), but about a week ago, I tried rebuilding a project (http://python-control.readthedocs.org) and the tables no longer appear.  Looking at the logs, it appears the files generated/foo.rst and generated/bar.rst are still being generated, just not added to the toctree.

Here is a log from a build that worked:
https://readthedocs.org/builds/python-control/2631561/
and one that no longer works:
https://readthedocs.org/builds/python-control/2860211/

**Comment by ericholscher** [comment]
This doesn't seem like an RTD issue. I'm guessing it's a version mismatch or some other issue with Sphinx?

**Comment by cwrowley** [comment]
Could be, but the reason I would think it was a RTD issue is that the same files worked on RTD on Apr 26, but then not on May 13.  Did the Sphinx version on RTD change between these two dates?

**Comment by ggventurini** [comment]
My project has the same issue.

Locally `autosummary` works properly, but on RTD no tables are shown. Notice my tables are in line, so this does not seem to be a problem of files being not added to the toc tree.

Eg. the following partial code from my doc

```
.. autosummary::
    :nosignatures:
    synthesizeNTF
    clans
    synthesizeChebyshevNTF
    simulateDSM
    simulateSNR
    realizeNTF
    stuffABCD
    mapABCD
    scaleABCD
    calculateTF
    realizeNTF_ct
    mapCtoD
    evalTFP
```

produces this on RTD: 

![](https://i.imgur.com/UlVyeeC.png)

direct link: https://i.imgur.com/UlVyeeC.png

and this on my local machines:

![](https://i.imgur.com/JX80zLs.png)

direct link: https://i.imgur.com/JX80zLs.png

**Comment by agjohnson** [comment]
We rolled out Sphinx 1.3.1 support on 2015-05-04. This issue is more than likely a Sphinx 1.3 change. Can you confirm this, and that you produce this issue running Sphinx 1.3.1 locally @ggventurini @cwrowley?

**Comment by cwrowley** [comment]
I tried running Sphinx 1.3.1 locally, and now I reproduce the issue locally.  So yes, this is definitely an issue with Sphinx, not RTD.  In fact, I now see that sphinx-doc/sphinx#1891 describes the same issue, and I can also confirm that adding the patch in sphinx-doc/sphinx#1892 fixes my issue locally.

I'm still not sure how to work around this in RTD, besides just waiting for the next release of Sphinx, so if anybody has suggestions, that would be very helpful.

**Comment by ggventurini** [comment]
Thanks for looking into this!

About working around the issue, I rewrote my tables by hand. That's my
workaround.

Not pretty, but it works. I will revert the commit when RTD is upgraded to
the next version of sphinx.

Best,

GV

**Comment by jschueller** [comment]
hi there, it seems it's a duplicate of #1822

**Comment by agjohnson** [comment]
This seems to be a confirmed Sphinx issue. We'll plan on getting the next release out when it gets shipped.

**Comment by dschult** [comment]
Has anyone found a workaround for autosummary on RTD besides creating all the tables manually?
The one-line fix in sphinx works great. But its not clear when a new version of Sphinx will be released.

**Comment by cwrowley** [comment]
I have not found a workaround on RTD besides creating the tables manually, and I would still very much like to find one.

**Comment by dschult** [comment]
Here's a workaround to make autosummary work on RTD.  Set up Read-the-docs to use a virtual machine build, which allows system-wide libraries to save load time [in "advanced settings"]. Then create a "requirements.txt" file for pip to know what to install.  Instead of installing sphinx-1.3.1 use a line similar to:

    -e git://github.com/sphinx-doc/sphinx.git@stable#egg=Sphinx-origin_stable

This builds a python egg for sphinx from the github version of sphinx-1.3.1.  Note that while the pypi version 1.3.1 is not fixed, the developers did add the fix to the github sphinx-1.3.1.  This doesn't take too long and it seems to work fine.

The ability to do this is described in the pip docs and in [an old but helpful post.](http://codeinthehole.com/writing/using-pip-and-requirementstxt-to-install-from-the-head-of-a-github-branch/)
